### PR TITLE
MOB-2619: Adds Permissions#getByOrganizationId

### PR DESCRIPTION
**Pull request by smstromb**
## Why?
1. Coordinator module needs broken apart into submodules prior to v1 release
2. Contxt needs access to the new user permissions endpoint https://github.com/ndustrialio/contxt-coordinator/pull/63

## What changed?
1. Moved all root level coordinator functions into submodules:
```js
//Applications
coordinator.createFavoriteApplication()	-> coordinator.applications.addFavorite()
coordinator.getAllApplications() -> coordinator.applications.getAll()
coordinator.getFavoriteApplications() -> coordinator.applications.getFavorites()
coordinator.getFeaturedApplications() -> coordinator.applications.getFeatured()
coordinator.deleteFavoriteApplication()	-> coordinator.applications.removeFavorite()

//Organizations
coordinator.getOrganizationById() -> coordinator.organizations.get()
coordinator.getAllOrganizations() -> coordinator.organizations.getAll()

//Permissions
coordinator.getUserPermissionsMap()	-> coordinator.permissions.getByUserId()

//Users
coordinator.activateNewUser() -> coordinator.users.activate()
coordinator.getUser() -> coordinator.users.get()
coordinator.getUsersByOrganization() -> coordinator.users.getByOrganizationId()
coordinator.inviteNewUserToOrganization() -> coordinator.users.invite() 
coordinator.removeUserFromOrganization() -> coordinator.users.remove()
```

2. Added `coordinator.permissions.getByOrganizationId(organizationId)` for fetching all user permissions for an entire organization.

### CODE REVIEW

File: src/coordinator/permissions.spec.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    context('when the organization ID is provided', function() {
+      let expectedUsersPermissions;
+      let userPermissionFromServer;
+      let expectedOrganizationId;
+      let promise;
+      let request;
+      let toCamelCase;
+
+      beforeEach(function() {
+        expectedOrganizationId = faker.random.uuid();
```
Comment (greg5green) [inline src/coordinator/permissions.spec.js]: Fixture here?

File: src/coordinator/permissions.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   * @param {string} baseUrl The base URL provided by the parent module
+   */
+  constructor(sdk, request, baseUrl) {
+    this._baseUrl = baseUrl;
+    this._request = request;
+    this._sdk = sdk;
+  }
+
+  /**
+   * Gets a list of user permissions for an entire organization
```
Comment (greg5green) [inline src/coordinator/permissions.js]: Could you make this a little more explicit that it is all the permission models for each user in an organization? The way it is makes it a little more ambiguous to me

File: support/fixtures/factories/contxtUserPermissions.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      times(
+        faker.random.number({ min: 0, max: 15 }),
+        () => factory.build('contxtStack').id
+      ),
+    stacksImplicit: () =>
+      times(
+        faker.random.number({ min: 0, max: 15 }),
+        () => factory.build('contxtStack').id
+      ),
+    userId: () => faker.random.uuid()
```
Comment (greg5green) [inline support/fixtures/factories/contxtUserPermissions.js]: We are using something like `test|${faker.internet.password()` elsewhere in the fixtures -- i'd recommend just using the user fixture

File: support/fixtures/factories/contxtUserPermissions.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+const factory = require('rosie').Factory;
+const faker = require('faker');
+const times = require('lodash.times');
+
+factory
+  .define('contxtUserPermissions')
+  .option('fromServer', false)
+  .attrs({
+    applicationsExplicit: () =>
+      times(
```
Comment (greg5green) [inline support/fixtures/factories/contxtUserPermissions.js]: Fun fact I learned a couple weeks ago: You can do this with Array.from!
```
Array.from(
  { length: faker.random.number({ min: 0, max: 15}) },
  () => factory.build('contxtApplication').id
)
```

Not a request! Repeat! Not a request: Do you like it better this way? I can't decide
```
factory
  .buildList('contxtApplication', faker.random.number({ min: 0, max: 15}))
  .map(({ id }) => id)
```
```

File: src/coordinator/permissions.spec.js (hunk 1 above)
Comment (smstromb) [reply src/coordinator/permissions.spec.js]: Good catch :+1:

File: support/fixtures/factories/contxtUserPermissions.js (hunk 3 above)
Comment (smstromb) [reply support/fixtures/factories/contxtUserPermissions.js]: I can't believe I missed this :man_facepalming:

File: support/fixtures/factories/contxtUserPermissions.js (hunk 4 above)
Comment (smstromb) [reply support/fixtures/factories/contxtUserPermissions.js]: I was not aware of `Array.from()` thanks for sharing.

And yes, I prefer your way better. Keep's lodash out of it :+1:

**Pull request merged**